source/extensions/http/cache/file_system_http_cache/cache_file_fixed_block.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include <sys/types.h>
+
+#include <cstddef>
+#include <cstdint>
+
+#include "source/common/common/byte_order.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+namespace FileSystemHttpCache {
+
+// This represents a minimal header block on the cache entry file; it uses a struct that
+// is required to be packed, and explicit network byte order, to ensure consistency.
+class CacheFileFixedBlock {
+public:
+  CacheFileFixedBlock();
+
+  // Deserializes the string representation of a CacheFileFixedBlock into this instance.
+  void populateFromStringView(absl::string_view s);
+
+  // Returns the size in bytes of a CacheFileFixedBlock.
+  static size_t size() { return sizeof(contents_); }
+
+  // fileId is a fixed value used to identify that this is a cache file.
+  uint32_t fileId() const { return fromEndianness<ByteOrder::BigEndian>(contents_.file_id_); }
+
+  // cacheVersionId is a value that should be consistent between versions of the file
+  // cache implementation. Changing version in code will invalidate all cache
+  // entries where the version ID does not match.
+  uint32_t cacheVersionId() const {
+    return fromEndianness<ByteOrder::BigEndian>(contents_.cache_version_id_);
```

Comment:
this feels heavy -- why don't we transform the data when reading/writing it from files so it's always in native format in-memory?

If nothing else it makes debugging (with gdb) annoying to have the data mangled, especially if you store big-endian.

Reply:
This object is essentially the serialized buffer. Transforming on read/write would mean allocating another buffer.